docs/site/migration/auth/example.md

```diff
@@ -0,0 +1,640 @@
+---
+lang: en
+title: 'Migrating access control example'
+keywords: LoopBack 4.0, LoopBack 4, LoopBack 3, Migration
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/migration-auth-access-control-example.html
+---
+
+This example is migrated from
+[loopback-example-access-control](https://github.com/strongloop/loopback-example-access-control),
+and uses the authentication and authorization system in LoopBack 4 to implement
+the access control.
+
+# App Scenario
+
+In this example, we create "Startkicker" (a basic Kickstarter-like application)
+to demonstrate authentication and authorization mechanisms in LoopBack. The
+application consists of four types of users:
+
+- guest
+- owner
+- team member
+- administrator
+
+Each user type has permission to perform tasks based on their role and the
+application's ACL (access control list) entries.
+
+Here is the diagram that describes the models:
+
+![models](../../imgs/tutorials/access-control-migration/auth_example_scenario_model.png)
+
+And the following operations can be processed by different roles:
+
+- /projects/list-projects
+  - guest, admin, owner, teamMember
+- /projects/view-all
+  - admin
+- /projects/{id}/show-balance
+  - owner, teamMember
+- /projects/{id}/donate
+  - admin, owner, teamMember
+- /projects/{id}/withdraw
+  - owner
+
+This application includes some pre-created data. They are shown in the above
+diagram. For example, as you can see, u1 owns project1, and u1 created a team
+with u1 and u2 as members. This means u1 is the **owner** of project1, and u1
+and u2 are the **team members** of project1. And u3 is the **admin**.
+
+## Differences Between LoopBack 3 and 4
+
+LoopBack 3 has several built-in models that consists of a RBAC system. The
+models are `User`, `Role`, `RoleMapping`, `ACL`, and `AccessToken`. You can
+learn how they work together in the
+[Controlling data access](https://loopback.io/doc/en/lb3/Controlling-data-access.html)
+tutorial.
+
+The LoopBack 4 authorization system gives developers the flexibility to
+implement the RBAC on their own. You can leverage popular third-party libraries
+like [casbin](https://github.com/casbin/casbin) or [oauth0](https://auth0.com/)
```

Comment:
`oauth0` -> `oAuth0`.

Reply:
From that web site, it refers as `Auth0`. Should we use that instead?

Reply:
yeah double checked, I think `Auth0` is correct. oAuth is used for 2.0